alicloud/resource_alicloud_polardb_cluster.go

```diff
@@ -0,0 +1,454 @@
+			period := d.Get("auto_renew_period").(int)
+			request.Duration = strconv.Itoa(period)
+			request.PeriodUnit = string(Month)
+			if period > 9 {
+				request.Duration = strconv.Itoa(period / 12)
+				request.PeriodUnit = string(Year)
+			}
+		}
+
+		raw, err := client.WithPolarDBClient(func(polarDBClient *polardb.Client) (interface{}, error) {
+			return polarDBClient.ModifyAutoRenewAttribute(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		d.SetPartial("auto_renew")
+		d.SetPartial("renewal_status")
+		d.SetPartial("auto_renew_period")
+	}
+
+	if d.HasChange("maintain_time") {
+		request := polardb.CreateModifyDBClusterMaintainTimeRequest()
+		request.RegionId = client.RegionId
+		request.DBClusterId = d.Id()
+		request.MaintainTime = d.Get("maintain_time").(string)
+
+		raw, err := client.WithPolarDBClient(func(polarDBClient *polardb.Client) (interface{}, error) {
+			return polarDBClient.ModifyDBClusterMaintainTime(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		d.SetPartial("maintain_time")
+	}
+
+	if d.IsNewResource() {
+		d.Partial(false)
+		return resourceAlicloudPolarDBClusterRead(d, meta)
+	}
+
+	if d.HasChange("description") {
+		request := polardb.CreateModifyDBClusterDescriptionRequest()
+		request.RegionId = client.RegionId
+		request.DBClusterId = d.Id()
+		request.DBClusterDescription = d.Get("description").(string)
+
+		raw, err := client.WithPolarDBClient(func(polarDBClient *polardb.Client) (interface{}, error) {
+			return polarDBClient.ModifyDBClusterDescription(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		d.SetPartial("description")
+	}
+
+	if d.HasChange("security_ips") {
+		ipList := expandStringList(d.Get("security_ips").(*schema.Set).List())
+
+		ipstr := strings.Join(ipList[:], COMMA_SEPARATED)
+		// default disable connect from outside
+		if ipstr == "" {
+			ipstr = LOCAL_HOST_IP
+		}
+
+		if err := polarDBService.ModifyDBSecurityIps(d.Id(), ipstr); err != nil {
+			return WrapError(err)
+		}
+		d.SetPartial("security_ips")
+	}
+
+	d.Partial(false)
+	return resourceAlicloudPolarDBClusterRead(d, meta)
+}
+
+func resourceAlicloudPolarDBClusterRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	polarDBService := PolarDBService{client}
+
+	clusterAttribute, err := polarDBService.DescribePolarDBClusterAttribute(d.Id())
+	if err != nil {
+		if NotFoundError(err) || IsExceptedErrors(err, []string{InvalidDBClusterIdNotFound, InvalidDBClusterNameNotFound}) {
```

Comment:
The line 264 error comes from a service method and it can be `NotFounderror`. So, the line 266 should be `if NotFoundError(err) {`